### DISCUSSION TITLE
Authentication Strategy to Enable Microservice Architecture.

### DISCUSSION THREAD

**Discussion by schwannden** [fastapi-users/fastapi-users#1021] [Ideas]
If we can add an authentication strategy called `external-service`, that request JWT token from another service (that is also implemented with fastapi-user), this library could be a powerful tool to build microservices with centralized user management, The flow would be
1. Service A is the main service handling user registration, deletion, ....etc, and configure auth strategy to JWT, using mongo as backend. 
2. Service A has 2 transport available, cookie transport for its own frontend, and jwt transport for a service B to get login token.
3. Service B configure auth strategy to `external-service`, and obtain JWT from service A via service A's jwt login route.

**Comment by frankie567** [comment]
Hmm, I see. In this case, does Service B really needs to obtain JWT by itself?

It could just check if the provided JWT is valid (through the proper authentication backend), but actually don't provide any specific route.

When you need a token, you would always call Service A and then just use it in Service B (and more). What do you think?

**Reply by schwannden** [reply]
This is true indeed, the reason why we need service B to login is more nuanced.

In microservice development, we would let developer for service B to start service B on their own dev environment, and this user would use the generated swagger doc to test. If we let user go to service A, login, copy cookie value, come back to service B's swagger doc, open browser console, set cookie, this process is less friendly.

So I guess this is really more of a devops concern then an architectural improvement.
